manifests/charts/gateways/istio-egress/templates/deployment.yaml

```
@@ -263,7 +263,11 @@ spec:
 {{- if eq .Values.global.pilotCertProvider "istiod" }}
       - name: istiod-ca-cert
         configMap:
+{{- if and (hasKey .Values "experimental") .Values.experimental.caRootCertConfigMap }} 
+          name: {{ .Values.experimental.caRootCertConfigMap }}
```

```suggestion
          name: {{ .Values.caRootCertConfigMap }}
```

Why not do this and set `caRootCertConfigMap` to `istio-ca-root-cert` in `values.yaml`?

I'm not sure what we gain by making this convoluted.

---

I guess I agree that templating the name is not experimental per se- I just added it to the `experimental` section because it belongs to an experimental feature (multi-controlplane support w/ overlapping namespaces). If there's agreement on making it a first-class setting, I'm fine with that.

Another option would be to just include the revision name in the ConfigMap instead of making it configurable

---

> I guess I agree that templating the name is not experimental per se- I just added it to the experimental section because it belongs to an experimental feature (multi-controlplane support w/ overlapping namespaces). If there's agreement on making it a first-class setting, I'm fine with that.

Given how we treat helm values deprecation, I would very strongly be against creating a value key `experimental` for experimental features.

That will do nothing but create confusion and mess over time.

If the feature is experimental in the code, we should document the feature as experimental in the code, and not carve out a helm key path for it which we will have to deprecate and remove on a unique cadence.

---

> Another option would be to just include the revision name in the ConfigMap instead of making it configurable

Also works for me. Lots of charts have customizable prefixes/suffixes for resource names, if we call ours `revision` it works just as well as it does everywhere else.

---

> Another option would be to just include the revision name in the ConfigMap instead of making it configurable
> Also works for me. Lots of charts have a globally-customizable prefix/suffix used for all resource names in a chart, if we call ours revision it works just as well as it does everywhere else.

This would still need an opt-in or it would break existing users (or, at the very least, cause excessive churn of making many many useless configmaps).

Revisions were fundamentally designed to share a root of trust, would love to not change that without strong motivation

---

> Revisions were fundamentally designed to share a root of trust, would love to not change that without strong motivation

Yes, that's the downside of `istiod` being its own root of trust, which is our default mode - I agree changing that is probably not on the table in the short term.

But we can change ~literally everything else but that one special service account relatively easily.